### PR TITLE
add a new log material averaging option

### PR DESCRIPTION
**Pull request by Shangxin-Liu**
The log material averaging option is added. Similar to other averaging options, the test results have a narrower pressure range than no averaging. The pressure range of log averaging is also close to those of other averaging.

### CODE REVIEW

File: include/aspect/material_model/averaging.h [hunk 1]
```diff
@@ -55,6 +55,10 @@ namespace aspect
     * quadrature point to \f[ \bar x = \max_{1\le q\le Q} x_q \f] where
     * $x_q$ are the values at the $Q$ quadrature points.
     *
+    * - Log average: Set the values of each output quantity at every
+    * quadrature point to \f[ \bar x = 10^{\frac 1Q \sum_{q=1}^Q log10_{x_q}} \f]
```
Comment (bangerth) [inline include/aspect/material_model/averaging.h]: I think what you mean is `\log_{10} x_q`.

File: include/aspect/material_model/interface.h [hunk 2]
```diff
@@ -347,6 +347,10 @@ namespace aspect
        * points and computes the best bi- or trilinear approximation for them.
        * In other words, it projects the values into the $Q_1$ finite element
        * space. It then re-evaluate this projection at the quadrature points.
+       *
+       * - Log average: Set the values of each output quantity at every
+       * quadrature point to \f[ \bar x = 10^{\frac 1Q \sum_{q=1}^Q log10_{x_q}} \f]
+       * where $x_q$ are the values at the $Q$ quadrature points.
```
Comment (bangerth) [inline include/aspect/material_model/interface.h]: same here

**Comment by bangerth** [comment]
Looks generally ok, just with a couple of minor updates.

Any reason you do exponentiation and logarithms with respect to base 10, rather than using natural logs? It doesn't make any difference, of course, but would allow you to call `exp(...)` instead of `pow(10, ...)`.

**Comment by Shangxin-Liu** [comment]
Fixed the typo. Yeah, exp and pow(10,...) makes no difference for the average. Previously I normally use log10 for the order of viscosity value (convenient), so I use logarithms based on 10 here. @bangerth

**Comment by bangerth** [comment]
OK, thanks. Would you mind providing an entry to `doc/modules/changes.h` as a follow-up?

**Comment by Shangxin-Liu** [comment]
I create a new pull request to update the changes.h . It seems no way to add another commit for this closed one. @bangerth

**Pull request merged**